### ISSUE TITLE
Determine discrepancy between dcp1 and dcp2 catalogs

### ISSUE THREAD

**Issue by hannes-ucsc** [DataBiosphere/azul#2149]
For one, the number and overall size of files differs:

`dcp1`:

![image](https://user-images.githubusercontent.com/5143256/90050087-9284dd80-dc8a-11ea-88a4-f3d5756c0703.png)

`dcp2`:

![image](https://user-images.githubusercontent.com/5143256/90050256-cb24b700-dc8a-11ea-832f-7847e885fa28.png)

- [x] Determine what files are missing in the `dcp2` catalog.
- [x] Determine why they are missing. Did the adapter drop them? Did the importer. The indexer?
- [x] Determine if there are differences in the number of documents in the other aggregate and contribution indices
- [x] Compare the aggregate documents for ten randomly selected project and sample entities between the two catalogs

**Comment by hannes-ucsc** [comment]
I produce two compact manifests without any filters, one for each catalog and diffed the two. Three types of discrepancies occur:

A) Files missing in the dcp2 catalog
B) Files missing in the dcp1 catalog
C) Files with mangled unicode characters in entities associated with the file

I believe A and B are the result of a bug in Azul, see CW logs below.

I will post reproduction of C in the next comment.

```
**CloudWatch Logs Insights**  
region: us-east-1  
log-group-names: /aws/lambda/azul-indexer-dev, /aws/lambda/azul-indexer-dev-aggregate, /aws/lambda/azul-indexer-dev-contribute, /aws/lambda/azul-indexer-dev-aggregate_retry  
start-time: -86400s  
end-time: 0s  
query-string:
```
fields @timestamp, @message
| filter @message like 'a46089ce-335d-4847-857d-e65c8efc2616'
| sort @timestamp desc
| limit 20
```
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
|       @timestamp        |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              @message                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
|-------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| 2020-08-12 14:40:42.019 | [DEBUG] 2020-08-12T14:40:42.18Z fa551fc1-c075-5aa7-ba92-486fc1d7930a Successfully wrote contribution to files/a46089ce-335d-4847-857d-e65c8efc2616 by bundle ae2cbcc5-662d-4f48-bfc6-841f8458f375 at 2019-05-16T21:18:13.083000Z.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |
| 2020-08-12 14:40:42.018 | [INFO] 2020-08-12T14:40:42.18Z fa551fc1-c075-5aa7-ba92-486fc1d7930a PUT https://search-azul-index-dev-atlcgwfjbk6dvrtmiaskbs2swa.us-east-1.es.amazonaws.com:443/azul_v2_dev_dcp2_files/doc/a46089ce-335d-4847-857d-e65c8efc2616_ae2cbcc5-662d-4f48-bfc6-841f8458f375_2019-05-16T21%3A18%3A13.083000Z_exists?op_type=create&refresh=false [status:201 request:0.031s]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
| 2020-08-12 14:35:56.462 | [WARNING] 2020-08-12T14:35:27.281Z 80912483-4647-5642-a446-732d4620ed7b Failures: defaultdict(<class 'int'>, {ContributionCoordinates(entity=EntityReference(entity_type='files', entity_id='a46089ce-335d-4847-857d-e65c8efc2616'), aggregate=False, bundle=BundleFQID(uuid='ae2cbcc5-662d-4f48-bfc6-841f8458f375', version='2019-05-16T21:18:13.083000Z'), deleted=False): 1, ContributionCoordinates(entity=EntityReference(entity_type='projects', entity_id='74b6d569-3b11-42ef-b6b1-a0454522b4a0'), aggregate=False, bundle=BundleFQID(uuid='ae2cbcc5-662d-4f48-bfc6-841f8458f375', version='2019-05-16T21:18:13.083000Z'), deleted=False): 1})                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
| 2020-08-12 14:35:27.021 | [WARNING] 2020-08-12T14:35:27.21Z 80912483-4647-5642-a446-732d4620ed7b There was a general error with document ContributionCoordinates(entity=EntityReference(entity_type='files', entity_id='a46089ce-335d-4847-857d-e65c8efc2616'), aggregate=False, bundle=BundleFQID(uuid='ae2cbcc5-662d-4f48-bfc6-841f8458f375', version='2019-05-16T21:18:13.083000Z'), deleted=False): TransportError(429, 'circuit_breaking_exception', {'error': {'root_cause': [{'type': 'circuit_breaking_exception', 'reason': '[parent] Data too large, data for [<http_request>] would be [16448916377/15.3gb], which is larger than the limit of [16287753830/15.1gb], real usage: [16448910904/15.3gb], new bytes reserved: [5473/5.3kb]', 'bytes_wanted': 16448916377, 'bytes_limit': 16287753830}], 'type': 'circuit_breaking_exception', 'reason': '[parent] Data too large, data for [<http_request>] would be [16448916377/15.3gb], which is larger than the limit of [16287753830/15.1gb], real usage: [16448910904/15.3gb], new bytes reserved: [5473/5.3kb]', 'bytes_wanted': 16448916377, 'bytes_limit': 16287753830}, 'status': 429}). Total # of errors: 1, giving up.  |
| 2020-08-12 14:35:27.001 | [WARNING] 2020-08-12T14:35:27.1Z 80912483-4647-5642-a446-732d4620ed7b PUT https://search-azul-index-dev-atlcgwfjbk6dvrtmiaskbs2swa.us-east-1.es.amazonaws.com:443/azul_v2_dev_dcp2_files/doc/a46089ce-335d-4847-857d-e65c8efc2616_ae2cbcc5-662d-4f48-bfc6-841f8458f375_2019-05-16T21%3A18%3A13.083000Z_exists?op_type=create&refresh=false [status:429 request:0.019s]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
| 2020-08-12 07:03:18.533 | [INFO] 2020-08-12T07:03:18.532Z 95bf8199-72d1-5244-bdd8-d634e4a19718 PUT https://search-azul-index-dev-atlcgwfjbk6dvrtmiaskbs2swa.us-east-1.es.amazonaws.com:443/azul_v2_dev_dcp1_files_aggregate/doc/a46089ce-335d-4847-857d-e65c8efc2616?op_type=create&refresh=false&version_type=internal [status:201 request:0.195s]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |
| 2020-08-12 07:03:18.533 | [DEBUG] 2020-08-12T07:03:18.533Z 95bf8199-72d1-5244-bdd8-d634e4a19718 Successfully wrote aggregate for dcp1/files/a46089ce-335d-4847-857d-e65c8efc2616 with 1 contribution(s).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
| 2020-08-12 07:03:15.828 | [DEBUG] 2020-08-12T07:03:15.828Z 95bf8199-72d1-5244-bdd8-d634e4a19718 Number of contributions selected for aggregation, by entity: {'bundles/ae2cbcc5-662d-4f48-bfc6-841f8458f375': 1, 'cell_suspensions/3e69c2a2-773c-4d51-b18f-a7f978bf0985': 1, 'cell_suspensions/d980eed4-9ac6-4343-81fc-647b0d88239f': 18, 'files/313febdb-20ef-4774-971c-0c9c5e68a59f': 1, 'files/a46089ce-335d-4847-857d-e65c8efc2616': 1, 'files/aafe8f12-877b-42b5-9ade-2ab77d09bc5c': 1, 'files/ad6052b8-5482-4265-94fb-0234181655a9': 1, 'files/ed46748c-76be-40fd-93b7-ad5ebdef2a6a': 1, 'samples/2697ff81-136f-45bb-b29e-df094766b3ee': 166, 'samples/e4ce8567-7522-4b05-a5b4-8f5063c98db9': 136}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
| 2020-08-12 07:03:15.769 | [DEBUG] 2020-08-12T07:03:15.769Z 95bf8199-72d1-5244-bdd8-d634e4a19718 Number of contributions read, by entity: {'bundles/ae2cbcc5-662d-4f48-bfc6-841f8458f375': 1, 'cell_suspensions/3e69c2a2-773c-4d51-b18f-a7f978bf0985': 1, 'cell_suspensions/d980eed4-9ac6-4343-81fc-647b0d88239f': 18, 'files/313febdb-20ef-4774-971c-0c9c5e68a59f': 1, 'files/a46089ce-335d-4847-857d-e65c8efc2616': 1, 'files/aafe8f12-877b-42b5-9ade-2ab77d09bc5c': 1, 'files/ad6052b8-5482-4265-94fb-0234181655a9': 1, 'files/ed46748c-76be-40fd-93b7-ad5ebdef2a6a': 1, 'samples/2697ff81-136f-45bb-b29e-df094766b3ee': 166, 'samples/e4ce8567-7522-4b05-a5b4-8f5063c98db9': 136}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
| 2020-08-12 07:03:12.821 | [INFO] 2020-08-12T07:03:12.821Z 95bf8199-72d1-5244-bdd8-d634e4a19718 Aggregating 1 contribution(s) to entity dcp1/files/a46089ce-335d-4847-857d-e65c8efc2616                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
| 2020-08-12 07:03:12.820 | [INFO] 2020-08-12T07:03:12.820Z 95bf8199-72d1-5244-bdd8-d634e4a19718 Attempt 1 of handling 1 contribution(s) for entity dcp1/files/a46089ce-335d-4847-857d-e65c8efc2616                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |
| 2020-08-12 07:02:58.989 | [INFO] 2020-08-12T07:02:58.988Z 9dcc46fd-522b-568a-9a59-1150c4740bf3 PUT https://search-azul-index-dev-atlcgwfjbk6dvrtmiaskbs2swa.us-east-1.es.amazonaws.com:443/azul_v2_dev_dcp1_files/doc/a46089ce-335d-4847-857d-e65c8efc2616_ae2cbcc5-662d-4f48-bfc6-841f8458f375_2019-05-16T211813.083000Z_exists?op_type=create&refresh=false [status:201 request:0.314s]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
| 2020-08-12 07:02:58.989 | [DEBUG] 2020-08-12T07:02:58.989Z 9dcc46fd-522b-568a-9a59-1150c4740bf3 Successfully wrote contribution to files/a46089ce-335d-4847-857d-e65c8efc2616 by bundle ae2cbcc5-662d-4f48-bfc6-841f8458f375 at 2019-05-16T211813.083000Z.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2020-08-12 07:02:57.431 | [DEBUG] 2020-08-12T07:02:57.293Z 9dcc46fd-522b-568a-9a59-1150c4740bf3 Loading file a46089ce-335d-4847-857d-e65c8efc2616, version 2019-05-16T175147.531000Z.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
| 2020-08-12 07:02:57.431 | [DEBUG] 2020-08-12T07:02:57.431Z 9dcc46fd-522b-568a-9a59-1150c4740bf3 Loading object files/a46089ce-335d-4847-857d-e65c8efc2616.2019-05-16T175147.531000Z from bucket org-hca-dss-prod                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

**Comment by hannes-ucsc** [comment]
@raaidbroad 

I think Unicode characters present in the metadata are being mangled by the importer.

This command extracts the offending field from the staging area:

```
$ gsutil cat gs://azul-dss-adapter-dev/dev/metadata/project/8185730f-4113-40d3-9cc3-929271784c2b* | jq -a '.contributors[21].laboratory'
"Garvan\u2010Weizmann Centre for Cellular Genomics"
```

Note the Unicode hyphen `\u2010`, probably the result of the original metadata being authored in Excel which tends to do this. Nevertheless, it's valid JSON. In this case the JSON uses a Unicode escape sequence rather than a UTF-8 two or three byte sequence. Both must be supported, since both are valid JSON.

Without `-a` the hypen is printed

```
$ gsutil cat gs://azul-dss-adapter-dev/dev/metadata/project/8185730f-4113-40d3-9cc3-929271784c2b* | jq '.contributors[21].laboratory'
"Garvan‐Weizmann Centre for Cellular Genomics"
```

So far so good. In BQ that Unicode character in the JSON appears to be mangled:

```
$ bq query --project_id=broad-jade-dev-data --format=prettyjson '#standardSQL
SELECT content FROM `broad-jade-dev-data.dcp1_metadata_no_files.project` WHERE project_id = "8185730f-4113-40d3-9cc3-929271784c2b"
' | jq -r .[0].content | jq -a '.contributors[21].laboratory'
Waiting on bqjob_r21a8bade17530aed_00000173e5e609f1_1 ... (0s) Current status: DONE   
"Garvan\ufffd\ufffd\ufffdWeizmann Centre for Cellular Genomics"
```

Note the three repetions of `\ufffd`.

I don't want to rule out the possibility that I've messed the command lines up, especially the last one. Or that I am not using the correct query. In the Google Cloud Console it looks mangled, too:

![image](https://user-images.githubusercontent.com/5143256/90092485-e79e0f00-dcdd-11ea-94ff-f9304ca24aee.png)

**Comment by hannes-ucsc** [comment]
A) and B) should be fixed by #2159

**Comment by hannes-ucsc** [comment]
Fix for #2159 (PR #2160) has been deployed and reindexing finished. The row discrepancies A) and B) above are fixed. There are no discrepancies in entity counts between the two catalogs.

**Comment by raaidbroad** [comment]
C) should be fixed by https://github.com/DataBiosphere/hca-ingest/pull/27 the pipeline needs to rerun on the DCP/1 metadata and re-imported to the TDR now. I know the fix looks like it isn't changing anything, but believe me, it does hahah
